compiler/rustc_trait_selection/src/traits/chalk_fulfill.rs

```diff
@@ -41,7 +48,9 @@ impl<'tcx> TraitEngine<'tcx> for FulfillmentContext<'tcx> {
         infcx: &InferCtxt<'tcx>,
         obligation: PredicateObligation<'tcx>,
     ) {
-        assert!(!infcx.is_in_snapshot());
+        if !self.usable_in_snapshot {
```

Comment:
Why add this instead of delegating the InferCtxt.usable_in_snapshot?

Reply:
what do you mean with this? an `InferCtxt` is always usable in a snapshot :thinking: